code/modules/paperwork/clipboard.dm

```diff
@@ -8,119 +11,172 @@
 	w_class = WEIGHT_CLASS_SMALL
 	throw_speed = 3
 	throw_range = 7
-	var/obj/item/pen/haspen //The stored pen.
-	var/obj/item/paper/toppaper //The topmost piece of paper.
 	slot_flags = ITEM_SLOT_BELT
 	resistance_flags = FLAMMABLE
+	/// The stored pen
+	var/obj/item/pen/pen
+	/**
+	 * Weakref of the topmost piece of paper
+	 *
+	 * This is used for the paper displayed on the clipboard's icon
+	 * and it is the one attacked, when attacking the clipboard.
+	 * (As you can't organise contents directly in BYOND)
+	 */
+	var/datum/weakref/toppaper_ref
 
 /obj/item/clipboard/suicide_act(mob/living/carbon/user)
 	user.visible_message("<span class='suicide'>[user] begins putting [user.p_their()] head into the clip of \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
-	return BRUTELOSS//the clipboard's clip is very strong. industrial duty. can kill a man easily.
+	return BRUTELOSS //The clipboard's clip is very strong. Industrial duty. Can kill a man easily.
 
 /obj/item/clipboard/Initialize()
 	update_appearance()
 	. = ..()
 
 /obj/item/clipboard/Destroy()
-	QDEL_NULL(haspen)
-	QDEL_NULL(toppaper) //let movable/Destroy handle the rest
+	QDEL_NULL(pen)
 	return ..()
 
+/obj/item/clipboard/examine()
+	. = ..()
+	if(pen)
+		. += "<span class='notice'>Alt-click to remove [pen].</span>"
+	var/obj/item/paper/toppaper = toppaper_ref == null ? null : toppaper_ref.resolve()
+	if(toppaper)
```

Comment:
Would this not be the same as just checking `if(toppaper_ref !0 null)` or `if(toppaper_ref)`?
Nope, cuz A dead reference could exist, that gets only detected, when `.resolve()`d, right?

Reply:
You can never get something that's qdeleted from `.resolve()`. If it was QDELETED, resolve() returns null.

Reply:
Yeah, but I mean that you cannot simplify the resolve away, right? You need the `.resolve()` to check that the still-saved weakref's reference actually doesn't exist any more.
So `if(toppaper_ref != null)` and `if(toppaper_ref)` cannot replace `if(toppaper_ref?.resolve())`. If I understand this, at least.